### ISSUE TITLE
Sticky Scroll in Search panel becomes blank on deeply nested results

### ISSUE THREAD

**Issue by IllusionMH** [microsoft/vscode#202665]
Type: <b>Bug</b>

Sticky Scroll area becomes blank when I scroll through deeply nested results and cover a lot of space (I use it in Panel area). At first, I was thinking it's related to container height (as resize might hide content if height reduced) but same behavior reproducible in Side Bar as well.

Steps to reproduce:

1. Search for term that would produce deeply nester results e.g. `toolbar` in VS Code repo
1. Scroll down to encounter deeply nested results
1. Observe that Sticky Scroll area becomes blank but still takes a lot of space
1. Type to increase panel height, and scroll back and forth to have Sticky Scroll area properly rendered (but still relatively deeply nested)
1. Decrease panel height to see that Sticky Scroll area becomes blank again

<details>
<summary>Errors from DevTools console</summary>

```
  ERR D.recursiveMatchCount is not a function: TypeError: D.recursiveMatchCount is not a function
    at g.m (vscode-file://vscode-app/c:/Users/<user name>/AppData/Local/Programs/Microsoft%20VS%20Code%20Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:2638:16686)
    at g.renderCompressedElements (vscode-file://vscode-app/c:/Users/<user name>/AppData/Local/Programs/Microsoft%20VS%20Code%20Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:2638:15311)
    at y.renderElement (vscode-file://vscode-app/c:/Users/<user name>/AppData/Local/Programs/Microsoft%20VS%20Code%20Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:627:7274)
    at A.renderElement (vscode-file://vscode-app/c:/Users/<user name>/AppData/Local/Programs/Microsoft%20VS%20Code%20Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:626:36001)
    at K.p (vscode-file://vscode-app/c:/Users/<user name>/AppData/Local/Programs/Microsoft%20VS%20Code%20Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:626:53119)
    at K.setState (vscode-file://vscode-app/c:/Users/<user name>/AppData/Local/Programs/Microsoft%20VS%20Code%20Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:626:52275)
    at G.u (vscode-file://vscode-app/c:/Users/<user name>/AppData/Local/Programs/Microsoft%20VS%20Code%20Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:626:48699)
    at l.value (vscode-file://vscode-app/c:/Users/<user name>/AppData/Local/Programs/Microsoft%20VS%20Code%20Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:626:48235)
    at o.y (vscode-file://vscode-app/c:/Users/<user name>/AppData/Local/Programs/Microsoft%20VS%20Code%20Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:87:1902)
    at o.z (vscode-file://vscode-app/c:/Users/<user name>/AppData/Local/Programs/Microsoft%20VS%20Code%20Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:87:1972)
    at o.fire (vscode-file://vscode-app/c:/Users/<user name>/AppData/Local/Programs/Microsoft%20VS%20Code%20Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:87:2188)
    at l.value (vscode-file://vscode-app/c:/Users/<user name>/AppData/Local/Programs/Microsoft%20VS%20Code%20Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:190:49143)
    at o.y (vscode-file://vscode-app/c:/Users/<user name>/AppData/Local/Programs/Microsoft%20VS%20Code%20Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:87:1902)
    at o.fire (vscode-file://vscode-app/c:/Users/<user name>/AppData/Local/Programs/Microsoft%20VS%20Code%20Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:87:2119)
    at I.n (vscode-file://vscode-app/c:/Users/<user name>/AppData/Local/Programs/Microsoft%20VS%20Code%20Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:87:17588)
    at I.setScrollPositionNow (vscode-file://vscode-app/c:/Users/<user name>/AppData/Local/Programs/Microsoft%20VS%20Code%20Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:87:16586)
    at n.R (vscode-file://vscode-app/c:/Users/<user name>/AppData/Local/Programs/Microsoft%20VS%20Code%20Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:190:52820)
    at HTMLDivElement.$ (vscode-file://vscode-app/c:/Users/<user name>/AppData/Local/Programs/Microsoft%20VS%20Code%20Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:190:51746)
```

not sure if logs below are relevan as they might appear without an error from above

```
event.ts:887 [d23] potential listener LEAK detected, having 499 listeners already. MOST frequent listener (229):
check @ event.ts:887
q @ event.ts:1058
C @ menuService.ts:391
q @ event.ts:1066
n @ toolbar.ts:335
j @ instantiationService.ts:119
createInstance @ instantiationService.ts:85
renderTemplate @ searchResultsView.ts:138
renderTemplate @ objectTree.ts:118
renderTemplate @ abstractTree.ts:406
p @ abstractTree.ts:1686
setState @ abstractTree.ts:1635
u @ abstractTree.ts:1331
(anonymous) @ abstractTree.ts:1287
y @ event.ts:1148
z @ event.ts:1159
fire @ event.ts:1183
(anonymous) @ scrollableElement.ts:203
y @ event.ts:1148
fire @ event.ts:1179
n @ scrollable.ts:393
setScrollPositionNow @ scrollable.ts:303
R @ scrollableElement.ts:456
$ @ scrollableElement.ts:369

event.ts:888 Error
    at i.create (event.ts:901:25)
    at a.q [as onDidChange] (event.ts:1057:34)
    at Object.C [as onWillAddFirstListener] (menuService.ts:391:34)
    at c.q [as onDidChange] (event.ts:1066:20)
    at new n (toolbar.ts:335:24)
    at b.j (instantiationService.ts:119:18)
    at b.createInstance (instantiationService.ts:85:18)
    at g.renderTemplate (searchResultsView.ts:138:56)
    at y.renderTemplate (objectTree.ts:118:30)
    at A.renderTemplate (abstractTree.ts:406:38)
    at K.p (abstractTree.ts:1686:33)
    at K.setState (abstractTree.ts:1635:42)
    at G.u (abstractTree.ts:1331:16)
    at l.value (abstractTree.ts:1287:46)
    at o.y (event.ts:1148:13)
    at o.z (event.ts:1159:9)
    at o.fire (event.ts:1183:9)
    at l.value (scrollableElement.ts:203:19)
    at o.y (event.ts:1148:13)
    at o.fire (event.ts:1179:9)
    at I.n (scrollable.ts:393:18)
    at I.setScrollDimensions (scrollable.ts:268:8)
    at n.setScrollDimensions (scrollableElement.ts:290:20)
    at m.layout (listView.ts:828:26)
    at ee.layout (listWidget.ts:1657:13)
    at _.layout (abstractTree.ts:2703:13)
    at De.Fd (searchView.ts:1239:13)
    at De.X (searchView.ts:1245:8)
    at De.layout (paneview.ts:302:9)
    at u.layout (splitview.ts:285:14)
    at n.U (splitview.ts:1361:13)
    at n.layout (splitview.ts:877:8)
    at l.layout (paneview.ts:593:18)
    at R.layout (viewPaneContainer.ts:637:18)
    at z.layout (panecomposite.ts:56:27)
    at m.layout (compositePart.ts:487:25)
    at m.layout (paneCompositePart.ts:448:9)
    at m.layout (panelPart.ts:175:9)
    at n.y (gridview.ts:927:13)
    at n.layout (gridview.ts:910:8)
    at c.layout (splitview.ts:285:14)
    at n.U (splitview.ts:1361:13)
    at n.M (splitview.ts:989:8)
    at o.y (event.ts:1148:13)
    at o.fire (event.ts:1179:9)
    at l.value (event.ts:118:84)
    at o.y (event.ts:1148:13)
    at o.fire (event.ts:1179:9)
    at l.j [as value] (sash.ts:572:22)
    at o.y (event.ts:1148:13)
    at o.fire (event.ts:1179:9)
    at y (event.ts:40:41)
```

</details>

<details>
<summary>Recording (18.5 MB)</summary>

https://github.com/microsoft/vscode/assets/1312662/7471971a-c677-4118-b302-7653d8ee091d

</details>

VS Code version: Code - Insiders 1.86.0-insider (aed5f4d65d2dbfda2c98abd7eb371751016838ab, 2024-01-17T07:04:28.837Z)
OS version: Windows_NT x64 10.0.22631
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz (8 x 2995)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|31.50GB (16.03GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details>Extensions: none

**Comment by IllusionMH** [comment]
@aiday-mar just to double label check should be with `editor-sticky-scroll` or `tree-sticky-scroll`? My expectation was later.

**Comment by aiday-mar** [comment]
After rereading your issue, it appears this may be related to the tree sticky scroll instead. Pinging @benibenj about it.

**Comment by benibenj** [comment]
Thanks for reporting this @IllusionMH. Newly we started to compress nodes when the maximum number of sticky nodes was reached. The bug was caused due to trying to compress nodes which are not compressible (for example files can't be compressed with folders currently). I fixed it in #202675

**Comment by IllusionMH** [comment]
Works as expected now. Thanks!
